### ISSUE TITLE
[Structure] Add trees

### ISSUE THREAD

**Issue by geekygirlsarah** [codethesaurus/codethesaur.us#682]
# Description

This issue is for adding a new structure for all languages that will let you compare a new aspect of a language. Ideally someone with many programming languages will help work on this.

This structure should be how to create and use various tree structures. This could include:

* What built-in libraries to include to add functionality (if needed or available)
* How to create basic tree objects
* What type of tree structure is created behind the scenes
* Does the tree self-balance
* How to add nodes (leaves) to the tree
* How to traverse the tree
* How to remove nodes 
* How to create various types of other trees (binary, ternary. n-ary, AVL, red-black, b-tree etc.)

Or other things. It should take some thought on all of the features that any programming language could have. Upon making this PR, expect it to have some revisions as people from various programming language backgrounds offer suggestions.

# Requirements

1. Look over the documentation at https://docs.codethesaur.us/thesaurus/add-new-structure/
2. Add a new file under `web/thesauruses/_meta` named whatever feels appropriate
3. Open up `web/thesauruses/meta_info.json` and add your new file to that file in the `structures` section.
4. Add in some categories, then add in the concepts under that
5. Feel free to use any `_meta` file as examples for how to do it
6. Make a PR and we can help test and review it

**Comment by maishaSupritee** [comment]
Hi, I'm participating in Hacktoberfest 2025 and would like to be assigned to this issue. 
I have experience working with tree data structure in Python, Java, and C++. Though I would probably like to tackle Python and Java first.

**Comment by geekygirlsarah** [comment]
Hi! I'd be happy to give it to you, but two thoughts:

1. You shouldn't submit a PR until after Oct 1 for it to count for Hacktoberfest.
2. This isn't for implementing trees in Python/Java/C++, but it's for defining what the pages would look like for trees in Code Thesaurus. So if you head to something like https://codethesaur.us/reference/?concept=functions&lang=java%3B17, this issue would be to define all the concepts on the left that eventually people define the right side in the languages.  You still want it if that's the case?

**Comment by maishaSupritee** [comment]
Yes thank you, I would still like to be assigned to this issue as I myself write coding blogs breaking concepts down so I love the idea behind codethesaurus. So would love to contribute!
And yes, I will be making the PR after October 1st.

**Comment by geekygirlsarah** [comment]
Great! It's yours!  Feel free to look at the docs at https://docs.codethesaur.us, and of course any of the other concept JSON files. Feel free to ask any questions here too!

**Comment by maishaSupritee** [comment]
Hi, I have completed this issue! Please feel free to take a look at my PR and let me know if there are any changes to be made. Thank you!

**Comment by geekygirlsarah** [comment]
Leaving a comment to keep it open

**Comment by maishaSupritee** [comment]
Hi, was wondering if you had the chance to look through my PR yet!
